AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Core/DataStoreList.swift

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+import Combine
+
+/// `DataStoreList<ModelType>` is a DataStore-aware custom `Collection` that is capable of loading
+/// records from the `DataStore` on-demand. This is specially useful when dealing with
+/// Model associations that need to be lazy loaded.
+///
+/// When using `DataStore.query(_ modelType:)` some models might contain associations
+/// with other models and those aren't fetched automatically. This collection keeps track
+/// of the associated `id` and `field` and fetches the associated data on demand.
+public class DataStoreList<ModelType: Model>: List<ModelType>, ModelListDecoder {
+
+    /// If the list represents an association between two models, the `associatedId` will
+    /// hold the information necessary to query the associated elements (e.g. comments of a post)
+    internal var associatedId: Model.Identifier?
```

Comment:
No need to specify `internal` if both getter and setter are internal
```suggestion
    var associatedId: Model.Identifier?
```